wgsl/index.bs

```diff
@@ -12172,6 +12294,56 @@ executing a [=statement/discard=] statement), execution of the quad may be
 terminated; however, such termination is not considered to produce [=uniform
 control flow|non-uniform control flow=].
 
+## Subgroups ## {#subgroups}
+
+A <dfn noexport>subgroup</dfn> is a set of invocations which concurrently
+execute a [=compute shader stage|compute=] or [=fragment shader
+stage|fragment=] shader stage [=entry point=], and which can efficiently share
+data and [[#subgroup-ops|collectively]] compute results.
+Each invocation in a compute or fragment shader belongs to exactly one subgroup.
+In a compute shader, each subgroup is a subset of a particular [=compute shader stage/workgroup=].
+In a fragment shader, a subgroup might contain invocations from multiple [=draw commands=].
+Each [=quad=] will be contained in a single subgroup.
+
+The <dfn noexport>subgroup size</dfn> is the maximum number of invocations in a subgroup.
+This value is accessible via the [=built-in values/subgroup_size=] built-in value.
+Subgroup size is a [=uniform value=] within a [=compute shader stage/workgroup=],
+but not within a [=draw command=].
```

Comment:
Does this imply that the uniformity analysis may need to be run twice, on user-defined functions that are used both by compute and fragment shaders, because `subgroup_size` will be uniform in the first case, but non-uniform in the second case?

Reply:
Because functions are analyzed in topological order I don't think the analysis would ever run twice. Consider:
```
fn foo(size : u32) {
  if (size == 16) {
    _ = subgroupBallot(true);
  }
}

@fragment
fn fragMain(@builtin(subgroup_size) size : u32) {
  // Uniformity error
  // Param size in foo is required to be uniform
  // subgroup_size is non-uniform in fragment shaders
  foo(size);
}

@compute @workgroup_size(64,1,1)
fn compMain(@builtin(subgroup_size) size : u32) {
  // No uniformity error
  // Param size in foo is required to be uniform
  // subgroup_size is uniform in compute shaders
  foo(size);
}
```

The built-in values are only ever in entry point parameter lists, so the rest of the analysis has completed and annotated which params/values need to be uniform.